app/actions/gfwclient.py

```diff
@@ -0,0 +1,585 @@
+    scan: float
+    track: float
+    acq_date: datetime
+    acq_time: str
+    satellite: str
+    confidence: str
+    version: str
+    bright_ti5: float
+    frp: float
+    daynight: str
+
+class IntegratedAlert(pydantic.BaseModel):
+    latitude: float
+    longitude: float
+    confidence_label: str = pydantic.Field(
+        ..., alias="gfw_integrated_alerts__confidence"
+    )
+    confidence: float = 0.0
+    num_clustered_alerts: int = 1
+    recorded_at: datetime = pydantic.Field(..., alias="gfw_integrated_alerts__date")
+
+    @pydantic.validator(
+        "recorded_at",
+        pre=True,
+    )
+    def sanitized_date(cls, val) -> datetime:
+        return datetime.strptime(val, "%Y-%m-%d").replace(tzinfo=timezone.utc)
+
+    @pydantic.root_validator
+    def compute_confidence(cls, values):
+        values["confidence"] = (
+            1.0 if values.get("confidence_label", "") in {"high", "highest"} else 0.0
+        )
+        return values
+
+class Geometry(pydantic.BaseModel):
+    type: str
+    coordinates: List[List[List[float]]]
+
+
+class CreatedGeostore(pydantic.BaseModel):
+    created_on: datetime
+    updated_on: datetime
+    gfw_geostore_id: str
+    gfw_geojson: Geometry
+    gfw_area__ha: float
+    gfw_bbox: List[float]
+
+    @pydantic.validator("created_on", "updated_on")
+    def clean_timestamp(val):
+        return val if val.tzinfo else val.replace(tzinfo=timezone.utc)
+
+    @pydantic.validator("gfw_geostore_id")
+    def simply_geostore_id(val):
+        # when we get a Geostore ID back, it's in standard form, but 
+        # when using it as a query parameter, it needs to in compact form.
+        return val.replace('-', '').lower() if val else val
+
+
+class GeoStoreResponse(pydantic.BaseModel):
+    data: CreatedGeostore
+    status: str
+
+
+def giveup_handler(details):
+    d1, d2 = details['kwargs']['daterange']
+    d1 = d1.strftime("%Y-%m-%d")
+    d2 = d2.strftime("%Y-%m-%d")
+    logger.error(f"Failed to get alerts for dataset: {details['kwargs']['dataset']}, geostore_id:{details['kwargs']['geostore_id']}, daterange: ({d1} - {d2})")
+
+def backoff_hdlr(details):
+    print ("Backing off {wait:0.1f} seconds afters {tries} tries "
+           "calling function {target} with args {args} and kwargs "
+           "{kwargs}".format(**details))
+    
+
+class DataAPI:
+
+    DATA_API_URL = "https://data-api.globalforestwatch.org"
+    RESOURCE_WATCH_URL = "https://api.resourcewatch.org"
+    def __init__(self, *, username: str = None, password: str = None):
+
+        self._username = username
+        self._password = password
+        self._auth_gen = None
+        self._api_keys = []
+
+    @backoff.on_exception(backoff.expo, (httpx.TimeoutException, httpx.HTTPStatusError), max_tries=3)
+    async def get_access_token(self):
+
+        async with httpx.AsyncClient(timeout=httpx.Timeout(60.0, connect=3.1)) as client:
+            try:
+                response = await client.post(
```

Comment:
@marianobrc I could use your help in solving this.
My test has mocked this request using respx, but here it's throwing an `AllMockedAssertionError`. I haven't figured out why respx isn't resolving call.
The test is in `app/actions/test_gfwclient.py`.

Reply:
I think this is just a problem with how I have configured my mocks in the unit test.

Reply:
I discovered the error, I had decorated the test with `@respx.mock()` instead of `@respx.mock`.
It's working now.

Reply:
I see, you can use @respx.mock() too, and that's sometimes useful to customize settings, but then the function receives a `respx_mock` object as the first argument and you need to use that object to setup mocked routes. Also, the default behaviour is to assert that every request within the test was mocked and captured by respx. I guess that as the route wasn't set in the right object then you were getting `AllMockedAssertionError`. That default behaviour can also be changed.